### ISSUE TITLE
Don't ungroup groups with class names/IDs

### ISSUE THREAD

**Issue by slyrz** [RazrFalcon/svgcleaner#76]
This is just a suggestion. Consider the following SVG file:

```svg
<svg height="400" width="400" xmlns="http://www.w3.org/2000/svg">
  <g class="outer">
    <path class="inner" d="m100 100h200v200h-200z"/>
  </g>
</svg>
```

Currently `svgcleaner --remove-unresolved-classes false --indent 2 inp.svg out.svg` produces

```svg
<svg height="400" width="400" xmlns="http://www.w3.org/2000/svg">
  <path class="inner" d="m100 100h200v200h-200z"/>
</svg>
```

The group is gone though you were telling *svgcleaner* not to remove unresolved
classes. And if `path` doesn't have a class name in the input file, 
it produces

```svg
<svg height="400" width="400" xmlns="http://www.w3.org/2000/svg">
  <path class="outer" d="m100 100h200v200h-200z"/>
</svg>
```

This one might seem a bit more harmless, but sometimes you really want the
group to stay in your SVG files, one example might be:

```svg
<g class="object">
  <path d="m100 100h200v200h-200z" transform="matrix(...)"/>
</g>
```

The group allows you to set or animate properties like transform relative to 
their actual values. So if you want to move the path 5px up, you can ignore
the matrix and write

```css
.object {
  transform: translateY(-5px);
}
```

in your style sheet. But if *svgcleaner* collapses the group, the computed style effectively becomes

```svg
<path d="m100 100h200v200h-200z" transform="translateY(-5px)"/>
```

which breaks the picture.

To put it all in a nutshell: both alterations are problematic if you rely on CSS to style or
animate your SVG elements. I think *svgcleaner* shouldn't ungroup groups if the user gave them
class names or IDs.

Your thoughts on this?

**Comment by RazrFalcon** [comment]
1. `<g class="outer">` shouldn't be removed. I'll investigate this.
1. `<path class="outer"` on other hand is kinda valid. Probably the `style` shouldn't be inheritable.
1. The major one. `svgcleaner` is designed for cleaning SVG files made by vector editors. And the main feature of them is that all the data are inside the file and available to svgcleaner. svgcleaner doesn't know anything about external dependencies of any kind (like css, class attribute, etc.). So it's hard to predict what can go wrong. That's why files with animations, scripting and conditional processing are ignored. So files with class/style/css should be ignored too, but it will cut out a lot of files. So there is a compromise.

As far as I understand, it's something that can't be 'fixed' completely. I, probably, can fix your issue, but eventually it will lead to a new one.

Basically: do not use svgcleaner on handwritten files.

**Comment by slyrz** [comment]
Yes, I know it's basically undecidable for you, I just opened the issue as a mere suggestion. For one user who's happy that the groups were left untouched, there'll be another user who's unhappy that the groups weren't removed...

**Comment by RazrFalcon** [comment]
Well, I'll look into it anyway. I think it can be done as part of `--remove-unresolved-classes`.
